src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/BuildPropertyPage/TreatWarningsAsErrorsValueProvider.cs

```diff
@@ -21,21 +21,19 @@ public TreatWarningsAsErrorsValueProvider(ITemporaryPropertyStorage temporaryPro
 
         public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
         {
-            if (StringComparers.PropertyLiteralValues.Equals(unevaluatedPropertyValue, "true"))
+            if (!bool.TryParse(unevaluatedPropertyValue, out bool value))
             {
-                // When setting this to "true", remove WarningsAsErrors
-                await defaultProperties.SaveValueIfCurrentlySetAsync(WarningsAsErrorsProperty, _temporaryPropertyStorage);
-                await defaultProperties.DeletePropertyAsync(WarningsAsErrorsProperty, dimensionalConditions);
-                await defaultProperties.RestoreValueIfNotCurrentlySetAsync(WarningsNotAsErrorsProperty, _temporaryPropertyStorage);
-            }
-            else
-            {
-                // When settings this to "false", remove WarningsNotAsErrors
-                await defaultProperties.SaveValueIfCurrentlySetAsync(WarningsNotAsErrorsProperty, _temporaryPropertyStorage);
-                await defaultProperties.DeletePropertyAsync(WarningsNotAsErrorsProperty, dimensionalConditions);
-                await defaultProperties.RestoreValueIfNotCurrentlySetAsync(WarningsAsErrorsProperty, _temporaryPropertyStorage);
+                return null;
             }
 
+            // When true, remove WarningsAsErrors. Otherwise, remove WarningsNotAsErrors.
+            string removePropertyName = value ? WarningsAsErrorsProperty : WarningsNotAsErrorsProperty;
+            string restorePropertyName = value ? WarningsNotAsErrorsProperty : WarningsAsErrorsProperty;
+
+            await defaultProperties.SaveValueIfCurrentlySetAsync(removePropertyName, _temporaryPropertyStorage);
+            await defaultProperties.DeletePropertyAsync(removePropertyName, dimensionalConditions);
+            await defaultProperties.RestoreValueIfNotCurrentlySetAsync(restorePropertyName, _temporaryPropertyStorage);
```

Comment:
Is there a change here, or just refactoring?

Reply:
The primary acquisition of `value` changed from being a string to a boolean. I realized each group of statements (within brackets) was the same. So, I just wrote it in a way where there was only 1 set of statements. I could go on a big philosophical tangent, but if you want me to change it back to if-else, let me know. Personally, I haven't written an if-else statement in several years unless I have to. I like collapsing forking decisions into expressions and assignments.

Reply:
I couldn't see a functional change, so wanted to double check this was just a refactoring and that I wasn't missing anything. It can be useful to split refactoring into separate commits or PRs to make review easier. Or if your PR has chunks of code that are moved without changes, call that out in the PR to make review easier.